src/state/features/application/actions.ts

```diff
@@ -61,13 +62,15 @@ const updateStatus = createAsyncThunk(
 
 const getCompanyDetailsWithAddress = createAsyncThunk(
   `${name}/companyDetailsWithAddress/get`,
-  async (applicationId: string) => {
+  async ({applicationId, dispatch}: {applicationId: string, dispatch: Dispatch}) => {
     try {
+      dispatch(setStatusCode({ errorCode: '' }))
       return await ApplicationApi.getInstance().getCompanyDetailsWithAddress(
         applicationId
       )
-    } catch (error: unknown) {
+    } catch (error: any) {
```

Comment:
don't use `try { ... } catch (error: any) {...}` here - instead convert this into a RTK Query apiSlice with built in error handling